relion/_parser/class2D.py

```diff
@@ -119,29 +106,17 @@ def _load_job_directory(self, jobdir):
             )
         return particle_class_list
 
-    def find_last_iteration(self, type):
-        file_list = list(self._basepath.glob("**/*.star"))
-        filename_list = [x.name for x in file_list]
+    def find_last_iteration(self, job_path, typename):
         filename = None
-        for x in filename_list:
-            if "run" in x:
-                if type in x:
-                    if self.has_numbers(x):
-                        nlist = []
-                        nlist.append(x)
-                        number_list = []
-                        number = int(x[6:9])
-                        number_list.append(number)
-                        number_list.sort()
-                        last_iteration_number = number_list[-1]
-
-                        filename = (
-                            "run_it"
-                            + str(last_iteration_number).zfill(3)
-                            + "_"
-                            + type
-                            + ".star"
-                        )
+        for file_candidate in job_path.glob("run*.star"):
+            if typename in file_candidate.name:
+                number_list = []
+                if self.has_numbers(file_candidate.name):
+                    number = int(file_candidate.name[6:9])
+                    number_list.append(number)
+                number_list.sort()
+                last_iteration_number = number_list[-1]
+                filename = f"run_it{last_iteration_number:03d}_{typename}.star"
         return filename
```

Comment:
`number_list` still has (at most) a single entry.

I'm wondering if this sort of function would be more useful:
```python
  def _final_data_and_model(self, job_path):
        number_list = [entry.stem[6:9] for entry in job_path.glob("run_it*.star")]
        last_iteration_number = max(int(n) for n in number_list if n.isnumeric(), default=0)
        if not last_iteration_number:
            raise ValueError(f"No result files found in {job_path}")
        data_file = job_path / f"run_it{last_iteration_number:03d}_data.star"
        model_file = job_path / f"run_it{last_iteration_number:03d}_model.star"
        for check_file in (data_file, model_file):
            if not check_file.exists():
                raise ValueError(f"File {check_file} missing from job directory")
        return data_file, model_file
```

This also has the benefit of only iterating once through the directory entries.